### DISCUSSION TITLE
How to search for Products/Variants based on CustomFields

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1525] [Q&A]
Look for the option to search for Product/Variant based on a value of custom fields.

**Comment by michaelbromley** [comment]
Hi! Searching by custom fields is not supported by the `search` query by default. It is possible to implement this using the ElasticsearchPlugin though, using a combination of [customMappings](https://www.vendure.io/docs/typescript-api/elasticsearch-plugin/elasticsearch-options/#customproductmappings) and [mapQuery](https://www.vendure.io/docs/typescript-api/elasticsearch-plugin/elasticsearch-options/#mapquery) options.
